File: LiveFromDAP/performance/scenario/binary_search/python/17.py
['a', 'b', 'c', 'd', 'e', 'f'], 'g';None;false
def binary_search(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = (left + right) // 2
        value = arr[mid]
        
        if value < target:
            left = mid + 1
        elif value > target:
            right = mid - 1
        else:
            return mid

File: LiveFromDAP/performance/scenario/binary_search/c/17.c
{'a', 'b', 'c', 'd', 'e', 'f'}, 6, 'g';-2;true
int binary_search(char arr[], int length, char target) {
    int left = 0;
    int right = length - 1;
    
    while(left <= right){
        int mid = (left + right) / 2;
        char value = arr[mid];

        if(value < target) {
            left = mid + 1;
        } else if(value > target) {
            right = mid - 1;
        } else {
            return mid;
        }
    }
    return -2; // random value otherwise
}